src/globus_sdk/utils.py

```diff
@@ -41,7 +45,10 @@ def safe_stringify(value):
     if isinstance(value, str):
         return value
     if isinstance(value, bytes):
-        return value.decode("utf-8")
+        try:
+            return value.decode("utf-8")
+        except UnicodeDecodeError:
+            raise ValueError("couldn't decode bytes as utf-8: {value}")
```

Comment:
This would need to be an f-string to render. But I don't think it's even desirable, since rendering bytes as part of an error about failure to render bytes is... confusing.

Why not just stick with the original line here, `value.decode("utf-8")`?

If it's important to reraise something else here, it should chain from the original error (`raise from`).